### PR TITLE
Docs: new index page

### LINKED ISSUE #342: Documentation: Improve landing page & split usage page
part of: https://github.com/stackabletech/documentation/issues/408

example of an improved landing page: https://docs.stackable.tech/home/stable/druid/index.html

The new landing page should feature:

- meta tags and a meta description
- an introduction of the tool
- a section about getting started
- a description and diagram of the resources it creates
- a description of its dependencies or other operators that interact with it
- a link to demos that use the operator
- the versions (as already present)

To split the usage guide, turn it into a section like [here](https://docs.stackable.tech/home/stable/druid/usage-guide/index.html). For pages that exist in Druid/other operators, try to keep the same ordering as in those operators.

### PR DESCRIPTION
**Pull request by fhennig**
# Description

fixes #342 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [ ] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: docs/modules/hive/pages/usage-guide/configuration-environment-overrides.adoc [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+All override property values must be strings. The properties will be formatted and escaped correctly into the XML file.
+
+For a full list of configuration options we refer to the Hive https://cwiki.apache.org/confluence/display/hive/configuration+properties[Configuration Reference].
+
+== The security.properties file
+
+The `security.properties` file is used to configure JVM security properties. It is very seldom that users need to tweak any of these, but there is one use-case that stands out, and that users need to be aware of: the JVM DNS cache.
+
+The JVM manages it's own cache of successfully resolved host names as well as a cache of host names that cannot be resolved. Some products of the Stackable platform are very sensible to the contents of these caches and their performance is heavily affected by them. As of version 3.1.3 Apache Hive performs poorly if the positive cache is disabled. To cache resolved host names, you can configure the TTL of entries in the positive cache like this:
```
Comment (maltesander) [inline docs/modules/hive/pages/usage-guide/configuration-environment-overrides.adoc:50]:
```suggestion
The JVM manages its own cache of successfully resolved host names as well as a cache of host names that cannot be resolved. Some products of the Stackable platform are very sensible to the contents of these caches and their performance is heavily affected by them. As of version 3.1.3 Apache Hive performs poorly if the positive cache is disabled. To cache resolved host names, you can configure the TTL of entries in the positive cache like this:
```

File: docs/modules/hive/pages/usage-guide/derby-example.adoc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      user: APP
+      password: mine
+      dbType: derby
+  metastore:
+    roleGroups:
+      default:
+        replicas: 1
+----
+
+WARNING: You should not use the `Derby` database with more than one replica or in production. Derby stores data locally and therefore the data is not shared between different metastore Pods and lost after Pod restarts.
```
Comment (maltesander) [inline docs/modules/hive/pages/usage-guide/derby-example.adoc:32]:
```suggestion
WARNING: You should not use the `Derby` database in production. Derby stores data locally which does not work in high availability setups (multiple replicas) and all data is lost after Pod restarts.
```

File: docs/modules/hive/pages/usage-guide/derby-example.adoc [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+To create a single node Apache Hive Metastore (v2.3.9) cluster with derby and S3 access, deploy a minio (or use any available S3 bucket):
```
Comment (maltesander) [inline docs/modules/hive/pages/usage-guide/derby-example.adoc:34]: The version is different from the example
```suggestion
To create a single node Apache Hive Metastore (v3.1.3) cluster with derby and S3 access, deploy a minio (or use any available S3 bucket):
```

**Review by maltesander — CHANGES_REQUESTED** [review]

**Comment by fhennig** [comment]
Thanks for the suggestions Malte! All added.

**Pull request merged**